File: sdks/python/src/raven/resources/connections.py
"""Real, event-sourced RTC connection data (Phase 9 observability), read from your backend."""

from __future__ import annotations

from typing import Any, cast

from .._async_http import AsyncRavenHttpClient
from .._http import RavenHttpClient
from .._types import ConnectionDetail, ConnectionSummary, ListConnectionsParams


def _list_query(params: ListConnectionsParams) -> dict[str, Any]:
    return {"roomId": params.room_id, "state": params.state, "limit": params.limit}


class ConnectionsResource:
    def __init__(self, http: RavenHttpClient) -> None:
        self._http = http

    def list(self, params: ListConnectionsParams | None = None) -> list[ConnectionSummary]:
        result = self._http.request("/v1/connections", query=_list_query(params or ListConnectionsParams()))
        return cast("list[ConnectionSummary]", result)

    def get(self, connection_id: str) -> ConnectionDetail:
        return cast(ConnectionDetail, self._http.request(f"/v1/connections/{connection_id}"))


class AsyncConnectionsResource:
    def __init__(self, http: AsyncRavenHttpClient) -> None:
        self._http = http

    async def list(self, params: ListConnectionsParams | None = None) -> list[ConnectionSummary]:
        result = await self._http.request("/v1/connections", query=_list_query(params or ListConnectionsParams()))
        return cast("list[ConnectionSummary]", result)

    async def get(self, connection_id: str) -> ConnectionDetail:
        return cast(ConnectionDetail, await self._http.request(f"/v1/connections/{connection_id}"))

File: packages/server-sdk/src/resources/connections.ts
import type { RavenHttpClient } from '../http-client';
import type { ConnectionDetail, ConnectionSummary, ListConnectionsParams } from '../types';

/** Real event-sourced RTC connection data (Phase 9 observability), read from your backend. */
export class ConnectionsResource {
  constructor(private readonly http: RavenHttpClient) {}

  list(params: ListConnectionsParams = {}): Promise<ConnectionSummary[]> {
    return this.http.request<ConnectionSummary[]>('/v1/connections', {
      query: { roomId: params.roomId, state: params.state, limit: params.limit },
    });
  }

  get(connectionId: string): Promise<ConnectionDetail> {
    return this.http.request<ConnectionDetail>(`/v1/connections/${connectionId}`);
  }
}
